pyscf/lib/libqcschema.py

```diff
@@ -0,0 +1,220 @@
+    data = None
+    temp = None
+    with open(file_name,'r') as f:
+        temp = json.load(f)
+    data = temp["trajectory"][-1]
+    return data
+
+def combine_syms_coords(syms,coords,to_Angstrom = False, xyz=False):
+    # QCSchema molecules are split into symbols and coordinates,
+    #  combine them here.
+    # Optionally convert from Bohr to Angstrom.
+    # Returns: either a geometry string (default) or
+    #  optionally it can return an xyz format string.
+    if(to_Angstrom):
+        # convert Bohr to Angstrom
+        coords = coords*BOHR
+        
+    NAtoms = len(syms)
+    geo = np.reshape(coords, (NAtoms,3))
+
+    # Concatenate the symbols and coordinates along the second axis
+    combined = np.concatenate([syms[:, np.newaxis], geo], axis=1)
+
+    # Convert the combined array to a string with spaces as separators
+    output = np.array2string(combined, separator=' ', max_line_width=np.inf, threshold=np.inf)
+
+    # Remove the brackets and quotes from the output string
+    PySCF_atoms = output.replace('[', '').replace(']', '').replace("'", '')
+
+    # Return as string or return as xyz-format string (i.e. top is NAtoms,blankline)
+    if(xyz):
+        xyz = f'{NAtoms}\n\n'
+        PySCF_atoms = xyz+PySCF_atoms
+    
+    return PySCF_atoms
+
+def load_qcschema_molecule(qcschema_dict, to_Angstrom = False, xyz=False):
+    # Load QCSchema molecule.
+    # Optionally convert geometry to angstrom
+    # Returns: either a geometry string (default) or
+    #  optionally it can return an xyz format string.
+
+    syms = np.array(qcschema_dict["molecule"]["symbols"])
+    geo = np.array(qcschema_dict["molecule"]["geometry"])
+
+    # combine together the symbols and coordinates
+    PySCF_atoms = combine_syms_coords(syms,geo,to_Angstrom,xyz)
+
+    return PySCF_atoms
+
+def load_qcschema_final_molecule(qcschema_dict, to_Angstrom = False, xyz=False):
+    # Load final molecule in QCSchema. 
+    # In GO job this is the optimized geometry.
+    # Optionally convert geometry to angstrom
+    # Returns: either a geometry string (default) or
+    #  optionally it can return an xyz format string.
+
+    syms = np.array(qcschema_dict["final_molecule"]["symbols"])
+    geo = np.array(qcschema_dict["final_molecule"]["geometry"])
+
+    # combine together the symbols and coordinates
+    PySCF_atoms = combine_syms_coords(syms,geo,to_Angstrom,xyz)
+
+    return PySCF_atoms
+
+def load_qcschema_hessian(qcschema_dict):
+    # qcschema_dict: pass in dict that has the qcschema output json loaded into it
+
+    # load qcschema hessian
+    qc_h = []
+    qc_h = qcschema_dict["return_result"]
+
+    # Get Number of atoms N
+    syms = np.array(qcschema_dict["molecule"]["symbols"])
+    NAtom = len(syms)
+
+    # reshape from (3N)**2 array to (N,N,3,3)
+    hessian = np.array(qc_h).reshape(NAtom,NAtom,3,3)
+    return hessian
+
+def load_qcschema_mol_scf(qcschema_dict,save_chk=False,chkfile="output.chk",to_Angstrom=False):
```

Comment:
build `mol` and `scf` objects in two functions. Do not mix the two in one function.